### PR TITLE
Update shovill module

### PR DESCRIPTION
**Pull request by rpetit3**
## PR checklist

Closes https://github.com/nf-core/modules/issues/329

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

This PR updates the shovill module (related issue https://github.com/nf-core/modules/issues/329)

### CODE REVIEW

**Comment by rpetit3** [comment]
Heads up @drpatelh I'm using your dsl2 branch to create the modules (e.g. `nf-core modules create . --tool shovill --author @rpetit3 --label shovill --meta`) and the linter doesn't like the line ending of the `functions.nf` file. 

An open and save on my end fixes the issue.

**Comment by grst** [comment]
> The linter doesn't like the line ending of the functions.nf file.

This is fixed in the latest release (1.13.2) of nf-core tools.

**Comment by FloWuenne** [comment]
I ran the pytest workflow tests locally using singularity and I can confirm the fails from the CI.

`TMPDIR=~ PROFILE=singularity pytest --tag shovill --symlink --keep-workflow-wd`

Digging into the shovill.logs it seems that shovill_spades fails due to memory issues:

```
[spades]   0:00:06.279    12M / 2G    ERROR   General                 (mmapped_reader.hpp        :  52)   mmap(2) failed. Reason: Cannot allocate memory. Error code: 12
[spades] 
[spades] 
[spades] == Error ==  system call for: "['/usr/local/bin/spades-core', '/home/florian/pytest_workflow_3oytn6p2/shovill_with_spades/work/17/ca0db7764f3db2eb7fd50cbb3e6ed1/spades/K37/configs/config.info', '/home/florian/pytest_workflow_3oytn6p2/shovill_with_spades/work/17/ca0db7764f3db2eb7fd50cbb3e6ed1/spades/K37/configs/isolate_mode.info']" finished abnormally, OS return value: 255
```

@rpetit3 Could you check out the settings for shovill_with_spades? I think this is the only one failing the pytest runs, I checked the other 3 shovill_* tests and they all pass.

**Comment by rpetit3** [comment]
Let's see how this goes, I'm forcing SPAdes to only use k=31 for assembly, can reduce it further is necessary.

I've also properly (e.g. used toGiga()) set the value for memory.

**Comment by rpetit3** [comment]
Everything except conda is passing. The version of SPAdes used by conda fails to reproduce the MD5s created by the containers. The containers are shipped with SPAdes v3.14.0

**Review by FloWuenne — APPROVED** [review]
LGTM! 

Conda failures when Docker and singularity passing are ok, according to @drpatelh !

File: software/shovill/main.nf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    } else {
+        container "quay.io/biocontainers/shovill:1.1.0--0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("contigs.fa"), emit: contigs
+    tuple val(meta), path("shovill.corrections"), emit: corrections
```
Comment (drpatelh) [inline software/shovill/main.nf:26]: Be great if we can line up these `emit` statements better for readability

File: software/shovill/main.nf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    tuple val(meta), path("shovill.log"), emit: log
+    tuple val(meta), path("{skesa,spades,megahit,velvet}.fasta"), emit: raw_contigs
```
Comment (drpatelh) [inline software/shovill/main.nf:28]: Can we have separate named channels for this?

**Review by drpatelh — CHANGES_REQUESTED** [review]
Looking great! Few comments.

**Comment by rpetit3** [comment]
Thank you very much for the feedback, think I'm getting a hang of it now.

**Review by drpatelh — APPROVED** [review]
👌🏽

**Pull request merged**